### PR TITLE
Adding action to sort issues

### PR DESCRIPTION
**Pull request by choldgraf**
This adds a little GitHub Action that does the following:

- Loops through each column of our organizational Project Boards
- For cards that point directly to GitHub Issues, grab the labels for each issue.
- For issues with a "priority" label (e.g. `prio: high`), *sort the issues* based on high- to low-priority
- Sort that board's column so that the high priority cards are at the top

This should make it easier for us to keep the high priority issues at high-visibility for the team! It runs each day on a CRON job

(note I also ran `black` on the scripts so some minor formatting has changed)

cc @damianavila who might think this is neat :-)

### CODE REVIEW

File: scripts/post-team-sync.py [hunk 1]
```diff
@@ -96,6 +100,12 @@
...
 # %%
 # Now create an issue with this content
-resp = api.issues.create("2i2c-org", "team-compass", title=f"Team Sync - {date.today():%b %d, %Y}", body=template, labels=metadata['labels'])
+resp = api.issues.create(
+    "2i2c-org",
+    "team-compass",
+    title=f"Team Sync - {date.today():%b %d, %Y}",
+    body=template,
+    labels=metadata["labels"],
+)
```
Comment (damianavila) [inline scripts/post-team-sync.py:109]: Using some new formatter? 😜

File: scripts/sort-issues.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+#       extension: .py
+#       format_name: hydrogen
+#       format_version: '1.3'
+#       jupytext_version: 1.10.3
+#   kernelspec:
+#     display_name: Python 3
+#     language: python
+#     name: python3
+# ---
+
```
Comment (damianavila) [inline scripts/sort-issues.py:15]: ahhhh... prototyping in the notebook 😉

File: scripts/sort-issues.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+                card["sort_label"] = label["name"]
+                cards_with_sorting_label.append(card)
+
+    # Sort the cards based on the sorting label
+    cards_sorted = sorted(
+        cards_with_sorting_label, key=lambda a: priority_list.index(a["sort_label"])
+    )
+
+    # Loop through the sorted cards (in reverse so we move the last label first) and move them to the top
+    for card in cards_sorted[::-1]:
```
Comment (damianavila) [inline scripts/sort-issues.py:72]:
```python
reversed(cards_sorted)
``` 
:wink:

File: scripts/sort-issues.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        print(f"\t\tMoved card {card['id']} to top")
+
+
+# Loop through our projects to sort cards
+for project in api.projects.list_for_org("2i2c-org"):
+    print(f"Sorting columns for project {project['name']}")
+
+    columns = api.projects.list_columns(project["id"])
+    for column in columns:
+        if any(ii in column["name"].lower() for ii in ["archive", "done"]):
```
Comment (damianavila) [inline scripts/sort-issues.py:83]: Is this over-complicated or am I missing something about the API response?
Is not enough to check if `column["name"].lower()` is in `["archive", "done"]` to `continue`?

**Comment by damianavila** [comment]
> cc @damianavila who might think this is neat :-)

I think it is neat, indeed!
I have already provided some feedback.

File: scripts/post-team-sync.py (hunk 1 above)
Comment (damianavila) [reply scripts/post-team-sync.py:109]: And now I saw your comment about black :wink:

File: scripts/sort-issues.py (hunk 4 above)
Comment (choldgraf) [reply scripts/sort-issues.py:83]: lol indeed that is a much more clear way of doing this 🤦‍♂️

File: scripts/sort-issues.py (hunk 3 above)
Comment (choldgraf) [inline scripts/sort-issues.py:72]:
```suggestion
    for card in reversed(cards_sorted):
```

File: scripts/sort-issues.py (hunk 4 above)
Comment (choldgraf) [reply scripts/sort-issues.py:83]: ah no nevermind, I knew there was a reason that I did it this way haha. The problem is that the columns aren't *just* named `archive` or `done`, so we just want to know if `archive` or `done` is _somewhere_ in the column name, but don't expect it to match the whole name

File: scripts/sort-issues.py (hunk 2 above)
Comment (choldgraf) [reply scripts/sort-issues.py:15]: indeed - it means you can open up this python script *as a notebook* with jupytext, which is helpful for modifications and real-time prototyping

File: scripts/sort-issues.py (hunk 4 above)
Comment (damianavila) [reply scripts/sort-issues.py:83]: OK, then a brief comment about it would be nice for the next reader.

File: scripts/sort-issues.py [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
```
Comment (damianavila) [inline scripts/sort-issues.py:83]:
```suggestion
        # columns aren't just named `archive` or `done`,
        # so we just want to know if `archive` or `done` is somewhere in the column name
        if any(ii in column["name"].lower() for ii in ["archive", "done"]):
```

**Comment by choldgraf** [comment]
thanks for the comment addition @damianavila - sorry I missed it before! let's see if this works :-)

**Pull request merged**